chgnet/graph/converter.py

```diff
@@ -59,13 +60,24 @@ def __init__(
         )
 
         # Set graph conversion algorithm
-        self._make_graph = None
-        if algorithm == "fast" and make_graph is not None:
-            self.create_graph = self._create_graph_fast
-            self.algorithm = "fast"
-        else:
-            self.create_graph = self._create_graph_legacy
-            self.algorithm = "legacy"
+        self.create_graph = self._create_graph_legacy
+        self.algorithm = "legacy"
+        if algorithm == "fast":
+            if make_graph is not None:
+                self.create_graph = self._create_graph_fast
+                self.algorithm = "fast"
+            else:
+                warnings.warn(
+                    "`fast` algorithm is not available, using `legacy`",
+                    UserWarning,
+                    stacklevel=1,
+                )
+        elif algorithm != "legacy":
+            warnings.warn(
+                f"Unknown algorithm {algorithm}, using `legacy`",
```

Comment:
Py3.8 introduced self-documenting f-strings:

```suggestion
                f"Unknown {algorithm=}, using `legacy`",
```

Reply:
Cool! I didn't know about this, thanks for pointing it out